lib/Queue.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * Queue => Stack
+ * -----------------------------
+ * Queue a stack data structure.
+ *
+ * A queue is a data structure in which the first element is inserted from one end called
+ * REAR (also called tail), and the deletion of existing element takes place from the other
+ * end called as FRONT (also called head).  This makes queue a FIFO (first in first out), or
+ * the element that is inserted first will always be the first to be removed.
+ *
+ * @private {object} _storage              Default value should be an empty object.
+ * @private {number} _start                Default value should be set 0.
+ * @private {number} _end                  Default value should be set 0.
```

Comment:
Cool second implementation! Updated the exercise prompt to say that the datastore object is optional and there is a way to finish the exercise without it.